magento/magento2#14664: Active state of shipping method not respecting store scope config in adminhtml

AleksLi opened the pull request:
<!---
    Thank you for contributing to Magento.
    To help us process this pull request we recommend that you add the following information:
     - Summary of the pull request,
     - Issue(s) related to the changes made,
     - Manual testing scenarios,
-->

<!--- Please provide a general summary of the Pull Request in the Title above -->

### Description
<!---
    Please provide a description of the changes proposed in the pull request.
    Letting us know what has changed and why it needed changing will help us validate this pull request.
-->
Added plugin to set Store in case of using the \Magento\Shipping\Model\Config\Source\Allmethods class somewhere in admin area. 

### Fixed Issues (if relevant)
<!---
    If relevant, please provide a list of fixed issues in the format magento/magento2#<[14664](https://github.com/magento/magento2/issues/14664)>.
    There could be 1 or more issues linked here and it will help us find some more information about the reasoning behind this change.
-->
1. magento-engcom/msi#<[14664](https://github.com/magento/magento2/issues/14664)>: Active state of shipping method not respecting store scope config in adminhtml

### Manual testing scenarios
<!---
    Please provide a set of unambiguous steps to test the proposed code change.
    Giving us manual testing scenarios will help with the processing and validation process.
-->
1. Go through each step on the core issue

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

AleksLi committed:
magento/magento2#14664: Active state of shipping method not respecting store scope config in adminhtml
Created Plugin to Set Admin chosen store

magento-engcom-team commented:
Hi @AleksLi. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

rogyar commented:
Hi @AleksLi. Thank you for your contribution. Could you provide more information, please, about a reason for using the plugin instead of modifying the original method directly?

AleksLi commented:
@rogyar I didn't know I can do that. I will do that then.

AleksLi committed:
magento/magento2#14664: Active state of shipping method not respecting store scope config in adminhtml
Removed Plugin and Write changes in AllMethods.php file

rogyar commented on app/code/Magento/Shipping/Model/Config/Source/Allmethods.php:
```diff
@@ -22,13 +29,16 @@ class Allmethods implements \Magento\Framework\Option\ArrayInterface
         \Magento\Framework\App\Config\ScopeConfigInterface $scopeConfig,
-        \Magento\Shipping\Model\Config $shippingConfig
+        \Magento\Shipping\Model\Config $shippingConfig,
+        RequestInterface $request
```
Please, fetch the `$request` instance via `objectManager` in case if it's passed as `null` (reference: "Adding a constructor parameter" section of https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/)

AleksLi replied:
Ok. Thanks. I will follow this example then.

rogyar commented on app/code/Magento/Shipping/Model/Config/Source/Allmethods.php:
```diff
@@ -65,4 +79,24 @@ public function toOptionArray($isActiveOnlyFlag = false)
+     *
+     * @return int
+     */
+    protected function getAdminStoreId()
```
Please, use `private` scope when introducing a new method

AleksLi replied:
ok. Thanks. I will update that.

rogyar commented on app/code/Magento/Shipping/Model/Config/Source/Allmethods.php:
```diff
@@ -65,4 +79,24 @@ public function toOptionArray($isActiveOnlyFlag = false)
+    {
+        $website = $this->request->getParam('website');
+        if ($website) {
+            return (int) $website;
```
I believe this approach for resolving store id is incorrect. A website id might not be equal store id (case: one website, 5 store views). In this approach, we return website id instead of store id.

AleksLi replied:
I will look for something that might handle that if magento has that implemented. I think it is.

AleksLi replied:
It is impossible to activate Shipping method on [StoreView level.](https://monosnap.com/file/welKKJjQ2opEsPxaOBDjrOGDSo1fNt) I've tried even with `showInStore="1"` in `system.xml` file.

AleksLi replied:
I guess I will check it without store request param and go back here with comment.

AleksLi committed:
magento/magento2#14664: Active state of shipping method not respecting store scope config in adminhtml
Fixed comments regarding the code style and Backward compatibility

AleksLi committed:
magento/magento2#14664: Active state of shipping method not respecting store scope config in adminhtml
Remove store view layer from getAdminStoreId method. Impossible to set Activate on StoreView level for shipping methods.

AleksLi commented:
I've removed StoreView check from AllMethods. It is impossible to set Active param for Shipping method on StoreView level in Admin panel.

rogyar committed:
Minor refactoring

rogyar commented:
@AleksLi I believe it's a good decision. Active state of a shipping/payment method should not rely on a store view but website (store view is not supposed to be scope for shipping/payment methods activation).

magento-engcom-team commented:
Hi @rogyar, thank you for the review.
[ENGCOM-5388](https://jira.corp.magento.com/browse/ENGCOM-5388) has been created to process this Pull Request

engcom-Delta commented:
:heavy_check_mark: QA passed

engcom-Foxtrot commented:
@AleksLi I've pushed a commit to your repo to fix static tests. But that comiit isn't applied to this PR. And static build still fails. I suspect that it happens because your repo forked from magento/inventory repo. Could you please fork repo from magento/magento2 repo/ and create a new PR? Thank you in advance.

sidolov commented:
@AleksLi , I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!
